src/Services/Azure/DataExplorer/DataExplorerService.cs

```diff
@@ -0,0 +1,315 @@
+        {
+            throw new Exception($"Could not retrieve URI for cluster '{clusterName}'");
+        }
+
+        return clusterUri!;
+    }
+
+    public async Task<List<JsonDocument>> QueryItems(
+        string clusterUri, 
+        string databaseName, 
+        string query, string? tenant = null, 
+        AuthMethod? authMethod = AuthMethod.Credential, 
+        RetryPolicyArguments? retryPolicy = null)
+    {
+        ValidateRequiredParameters(clusterUri, databaseName, query);
+
+        var kcsb = await CreateKustoConnectionStringBuilder(
+            clusterUri,
+            authMethod,
+            null,
+            tenant);
+            
+        using var queryProvider = KustoClientFactory.CreateCslQueryProvider(kcsb);
+        var reader = await queryProvider.ExecuteQueryAsync(databaseName, query, null);
+        var results = new List<JsonDocument>();
+
+        try
+        {
+            while (reader.Read())
+            {
+                var dict = new Dictionary<string, object?>();
+                for (int i = 0; i < reader.FieldCount; i++)
+                {
+                    dict[reader.GetName(i)] = reader.GetValue(i);
+                }
+                var json = JsonSerializer.SerializeToDocument(dict);
+                results.Add(json);
+            }
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Error executing query: {ex.Message}", ex);
+        }
+        
+        return results;
+    }
+
+    private async Task<KustoConnectionStringBuilder> CreateKustoConnectionStringBuilder(
+        string uri,
+        AuthMethod? authMethod,
+        string? connectionString = null,
+        string? tenant = null)
+    {
+        switch (authMethod)
+        {
+            case AuthMethod.Key:
+                throw new NotSupportedException("Data Explorer data plane does not support key-based authentication via ARM. Use AAD credential or connection string.");
+            case AuthMethod.ConnectionString:
+                if (string.IsNullOrEmpty(connectionString))
+                    throw new ArgumentNullException(nameof(connectionString));
+                return new KustoConnectionStringBuilder(connectionString);
+            case AuthMethod.Credential:
+            default:
+                var credential = await GetCredential(tenant);
+                var builder = new KustoConnectionStringBuilder(uri).WithAadAzureTokenCredentialsAuthentication(credential);
+                if (!string.IsNullOrEmpty(tenant))
+                {
+                    builder.Authority = $"https://login.microsoftonline.com/{tenant}";
+                }
+                return builder;
+        }
+    }
+
+    public static string GetClusterNameFromUri(string clusterUri)
+    {
+        ValidateRequiredParameters(clusterUri);
+
+        var uri = new Uri(clusterUri);
+        var host = uri.Host;
+        var clusterName = host.Split('.')[0];
```

Comment:
Unfourtently, this logic is incorrect. Uri's don't map to names directly, especially in virtual cluster environments like Fabric. The correct logic would be to match against the cluster details like we do in the reverse logic.